packages/clis/parser/game-files/scs-archive.ts

```diff
@@ -1,34 +1,132 @@
+import { assert, assertExists } from '@truckermudgeon/base/assert';
 import { Preconditions } from '@truckermudgeon/base/precon';
 import fs from 'fs';
 import { createRequire } from 'module';
+import type { BaseOf } from 'restructure';
 import * as r from 'restructure';
 import zlib from 'zlib';
-import { uint64le } from './restructure-helpers';
+import { logger } from '../logger';
+import { DdsHeader } from './dds-parser';
+import { MappedNumber, uint64le } from './restructure-helpers';
 const require = createRequire(import.meta.url);
 
 // eslint-disable-next-line @typescript-eslint/no-unsafe-call
 const { city64 } = require('bindings')('cityhash') as {
   city64: (s: string) => bigint;
 };
 
+// eslint-disable-next-line @typescript-eslint/no-unsafe-call
+const { gdeflate } = require('bindings')('gdeflate') as {
+  gdeflate: (inBuffer: ArrayBuffer, outBuffer: ArrayBuffer) => number;
+};
+
 const FileHeader = new r.Struct({
   magic: new r.String(4),
   version: r.int16le,
   salt: r.int16le,
   hashMethod: new r.String(4),
-  numEntries: r.int32le,
-  entriesOffset: r.int32le,
+  entryTableCount: r.uint32le,
```

Comment:
🤔 I should probably restore HashFS v1 support (in addition to adding HashFS v2 support). 

It might take a few months for 1.50 to officially launch, and I'd rather have this code merged before then. Also: it's probably friendlier to have a "latest 2 versions" support policy instead of a "latest version, only" policy.

Reply:
The general sentiment over on the forum seems to be that the Nebraska release will happen this month. 1.50 is all but certain to have been launched before then.

I suspect that at this point, v1 support is a case of [YAGNI](https://en.wikipedia.org/wiki/YAGNI).

Reply:
Ooooo... thanks for the heads-up. In that case, I'll continue on as I originally planned (with the much-easier to maintain "latest version, only" policy 😁)